### ISSUE TITLE
Handling of a single frame input path has changed in 1.0.0

### ISSUE THREAD

**Issue by justinfx** [sqlboy/fileseq#20]
The behaviour in parsing and handling single frame paths has changed between 0.5.1 and 1.0.0

```python
src = "/path/to/file_v2.exr"
fs = fileseq.FileSequence(src)
print fs.start(), fs.end(), fs.padding(), str(fs)

# 0.5.1
(0, 0, '', '/path/to/file_v2.exr')

# 1.0.0
(2, 2, '@', '/path/to/file_v2@.exr')
```

I feel it is incorrect to interpret this pattern as a sequence, when it is really a version. I believe the old pattern handling required a naming scheme like this for seqs:  "basename.frames.ext"

This change has caused some unexpected bugs when the str() rep, padding, and range values suddenly changed between versions.

**Comment by justinfx** [comment]
tagging @michaelmovies and @sqlboy

**Comment by sqlboy** [comment]
Yeah that shouldn't be recognized as a sequence, that would break a lot of things I would imagine.

**Comment by michaelmovies** [comment]
It appears that neither version has gotten this right, which makes sense as the way FileSequence should solve a given string is ambiguous.

For instance, this should not resolve as a sequence?

"/to/file2.exr"

... but this should?

"/to/file.2.exr"

... and this should?

"/to/file2.2.exr"

In version 0.5.1 all three of those resolve to a sequence with a length of 1, a padding of "", and a start and end frame of 0.  However it seems that the latter 2 should both be interpreted as having a start and end frame of 2, and that was what I was attempting to fix.

In version 1.0.0 all three resolve to a sequence with a length of 1, padding of "@", and a start and end frame of 0.  That appears to be more accurate in 2/3 cases, less accurate in 1.

Unfortunately this comes down to a test having been put in place for a single frame, no padding, but no explicit test for a single versioned file without a frame.

That said it hinges on the fact that neither version of fileseq explicitly says that a sequential frame number only occurs between two period characters ...they just look for numbers, and do not special case the version scheme.

So what should the decision be?  Revert that behavior in all the cases above?  Special case anything that follows the _v# pattern as version number (what then happens for users who don't recognize that as a normal version string)?

M  

--
END OF LINE

**Comment by justinfx** [comment]
That is not the result I get when I perform the test cases you outline. All three patterns resolve to a start and stop of 2.

You are correct that there is not any strict documentation outlining that a range pattern must occur between two periods (other than what the regular expressions actually did), but all in in it means that the 1.0.0 changes behavior in an unexpected way. The fact that it breaks down on identifying the common "v2" version pattern indicates that this change will be undesirable. 

However, I agree that there is a bit of a bug in the handling of single frame paths, such as `/to/file.2.exr` where `0.5.1` produces: `start:0, end:0, pad:[None] `, and I would think it should produce 2/2/@

All in all, it would be desirable to fix the initial bug you mentioned, while having a test case to ensure that it does not recognize a pattern as a sequence when it contains *no padding* and the numbers are not between periods. 

Once we have original behaviour again, we could discuss if this should be treated as a sequence: `/to/file_2.exr`. Meaning, a number not surrounded by alpha characters.

**Comment by michaelmovies** [comment]
I agree that the behavior on /to/file2.ext should not have changed, that's a regression.  I'll try and add a test tomorrow that can work on both 0.5.1 and 1.0.x.  

Matt, how does this rule sound?

"a number is a version (and thus not a frame) IF it is NOT preceded by a period AND no padding character exists in the string being passed to FileSequence"

That should be easy enough to fix.

Secondarily, 1.0.0 introduced the null FrameSet (FrameSet('')), which is required for set comparison operators on the FrameSet object.  This has led to some confusion because FrameSet('') (and FrameSet(None) although that's a bug and should fail) no longer raises a ParseException.  An empty FrameSet seems perfectly reasonable to me, in that it enables so much additional functionality, but it is obviously a confusing change for some.

Should FrameSet disallow direct creation of the null, and return some special object or NotImplemented or None when you do a set operation that results in an empty set?  Or does it make more sense to have the null remain, as it does now, but with some operations (start, end, InvertedFrameRange) that essentially make no sense?  I would think the latter, with some added sensible defaults for those methods in 1.0.1.

M  

--
END OF LINE

**Comment by sqlboy** [comment]
I believe your rule about the number being a version and not a frame is correct.

I seem to recall another case from maya where underscores are used. I forget what menu option its under, maybe in the render tab, its where you can define your output sequence naming convention.  There was an obscure one someone was using that was not name.range.ext.  I'm not sure if they've changed since then (was a couple years ago) or if the current code is handling that case properly.

I was discovered when findSequencesOnDisk didn't detect it, but it had to know how to parse the sequence string as well.

**Comment by justinfx** [comment]
This should be fixed as of v1.1.0
